### PR TITLE
BJS2-14250 - Star Wars Arena

### PR DESCRIPTION
**Pull request by ravshanbekn**
+ Created Robot class: play role of fighters
+ Created Battle class: compares two robots and returns winner
+ Created Main class: tested functionality

### CODE REVIEW

File: src/main/java/faang/school/godbless/starwars_arena/Battle.java
Comment (pablo023) [inline src/main/java/faang/school/godbless/starwars_arena/Battle.java:1]: Ну чёт нагородил(

Comment (pablo023) [reply src/main/java/faang/school/godbless/starwars_arena/Battle.java:1]: public Future<Robot> fight(Robot firstRobot, Robot secondRobot) {
        if (executorService.isShutdown()) {
            throw new IllegalStateException("We are not accepting any candidates for battle anymore");
        }
        log.info("We have two candidates: {} and {}", firstRobot.getName(), secondRobot.getName());
        return executorService.submit(() -> determineWinner(firstRobot, secondRobot));
    }

    public void closeCandidateAccept() {
        log.info("No more candidates for battle");
        executorService.shutdown();
    }

    private Robot determineWinner(Robot firstRobot, Robot secondRobot) {
        int firstRobotPower = firstRobot.calculatePower();
        int secondRobotPower = secondRobot.calculatePower();

        log.info("Robots are fighting...");
        simulateFight();

        return comparePower(firstRobot, secondRobot, firstRobotPower, secondRobotPower);
    }

    private void simulateFight() {
        try {
            Thread.sleep(FIGHT_DURATION);
        } catch (InterruptedException e) {
            Thread.currentThread().interrupt();
            log.error("The fight was interrupted", e);
        }
    }

    private Robot comparePower(Robot firstRobot, Robot secondRobot, int firstPower, int secondPower) {
        if (firstPower > secondPower) {
            return firstRobot;
        } else if (secondPower > firstPower) {
            return secondRobot;
        } else {
            return breakTie(firstRobot, secondRobot);
        }
    }

    private Robot breakTie(Robot firstRobot, Robot secondRobot) {
        boolean isFirstLucky = ThreadLocalRandom.current().nextBoolean();
        return isFirstLucky ? firstRobot : secondRobot;
    }
}

**Review by pablo023 — CHANGES_REQUESTED** [review]
Давай получше сделаем.

File: src/main/java/faang/school/godbless/starwars_arena/Battle.java
Comment (ravshanbekn) [reply src/main/java/faang/school/godbless/starwars_arena/Battle.java:1]: И правда, с декомпозицией стало лучше

**Pull request merged**